archive/indexer.go

```diff
@@ -54,60 +83,57 @@ func CreateIndexes(dir string) error {
 	return err
 }
 
-func IndexLogFile(path string) error {
-	subdir := path + zarExt
-	zdxName := "zdx:type:ip"
-	zdxPath := filepath.Join(subdir, zdxName)
-	// XXX remove without warning, should have force flag
-	zdx.Remove(zdxPath)
-
-	fmt.Printf("%s: indexing as %s\n", path, zdxPath)
-
-	file, err := os.Open(path)
-	if err != nil {
-		return err
-	}
-	defer file.Close()
-	reader, err := detector.LookupReader("bzng", file, resolver.NewContext())
-	if err != nil {
-		return err
-	}
-	table, err := indexTypeIP(reader)
+func Run(path string, rules []Rule) error {
+	subdir, err := archiveDir(path)
 	if err != nil {
 		return err
 	}
-	if table.Size() == 0 {
-		//XXX
-		return errors.New("nothing to index")
-	}
-	// make subdirectory for index if it doesn't exist
-	if err := os.Mkdir(subdir, 0755); err != nil {
-		if !os.IsExist(err) {
-			return err
-		}
+	var indexers []Indexer
+	for _, rule := range rules {
+		indexer := rule.NewIndexer(subdir)
+		indexers = append(indexers, indexer)
+		fmt.Printf("%s: creating index %s\n", path, indexer.Path())
```

Comment:
debugging leftover?

Reply:
Good pint.  This was intended as output of command but should be elsewhere so this can be restified.

We need to chanelize this as well as the hits ini zar find so I think I will put this in an issue and leave for a later PR.

Reply:
👍